### ISSUE TITLE
ERROR: Failed to merge reports - Mochawesome

### ISSUE THREAD

**Issue by Madvic90** [cypress-io/cypress#7949]
### Current behavior:

I've installed the Mochawesome reporter and all the needed plugins. Whenever I try to run a folder it keeps giving me an ERROR that Cypress failed to merge the reports. It looks like that the script does not create a file in the folder mocha.

It only creates a `report.json` in Mochareports.
![image](https://user-images.githubusercontent.com/59475147/87162474-467afd80-c2c6-11ea-96c4-7d9d7bfaeb8d.png)

this is a part of my `package.json`.

```json
{
  "scripts": {
    "cypress:open": "npm install && cypress open",
    "cypress:open:local": "npm install && cypress open --env configFile=local",
    "cypress:open:dev": "npm install && cypress open --env configFile=dev",
    "cypress:open:tst": "npm install && cypress open --env configFile=tst",
    "cypress:open:acc": "npm install && cypress open --env configFile=acc",
    "cypress:run": "npm install && cypress run",
    "cypress:run:local": "npm install && cypress run --browser chrome --headless --env configFile=local",
    "cypress:run:tst": "npm install && cypress run --browser chrome --headless --env configFile=tst",
    "cypress:run:acc": "npm install && cypress run --browser chrome --headless --env configFile=acc",
    "cypress:run:acc:teampark": "npm install && cypress run --browser chrome --headless --env configFile=acc --spec cypress/integration/team_park.js --config video=false",
    "clean:reports": "(if exist cypress\\reports (rmdir /S /Q cypress\\reports)) && mkdir cypress\\reports && mkdir cypress\\reports\\mocha  &&  mkdir cypress\\reports\\mochareports",
    "pretest": "npm run clean:reports",
    "scripts": "cypress run --spec cypress/integration/website/schooltrip/**/*",
    "combine-reports": "mochawesome-merge  cypress/reports/mocha/*.json > cypress/reports/report.json",
    "generate-report": "marge cypress/reports/mochareports/report.json -f report -o cypress/reports/mochareports",
    "posttest": "npm run combine-reports && npm run generate-report",
    "test": "npm run scripts || npm run posttest"
```

### Desired behavior:

What cypress needs to do according to all online explanation is to create a file in the Mocha and Mochareports and merge these files in order to get an HTML report.

Error given after a run:
```
> @ posttest C:\Users\iessadeq\Desktop\*********l\******-cypress-tests-master
> npm run combine-reports && npm run generate-report


> @ combine-reports C:\Users\iessadeq\Desktop\***********\*****-cypress-tests-master
> mochawesome-merge  cypress/reports/mocha/*.json > cypress/reports/report.json

ERROR: Failed to merge reports

Error: Pattern cypress/reports/mocha/*.json matched no report files

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @ combine-reports: `mochawesome-merge  cypress/reports/mocha/*.json > cypress/reports/report.json`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the @ combine-reports script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\iessadeq\AppData\Roaming\npm-cache\_logs\2020-07-10T13_54_21_097Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @ posttest: `npm run combine-reports && npm run generate-report`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the @ posttest script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\iessadeq\AppData\Roaming\npm-cache\_logs\2020-07-10T13_54_21_243Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @ test: `npm run scripts || npm run posttest`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the @ test script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\iessadeq\AppData\Roaming\npm-cache\_logs\2020-07-10T13_54_21_386Z-debug.log
```
### Test code to reproduce
```json
"reporter": "cypress-multi-reporters",
    "reporterOptions": {
        "reporterEnabled": "mochawesome",
        "mochawesomeReporterOptions": {
            "reportDir": "cypress/reports/mocha",
            "quite": true,
            "overwrite": false,
            "html": false,
            "json": true
        }
```

I've tried several things without any succes.. Could anyone help please?

### Versions
```json
  "devDependencies": {
    "chance": "^1.1.0",
    "cypress": "^4.7.0",
    "cypress-file-upload": "^3.5.3",
    "cypress-multi-reporters": "^1.4.0",
    "lodash": "^4.17.15",
    "mocha": "^8.0.1",
    "mocha-junit-reporter": "^1.23.3",
    "mochawesome": "^6.1.1",
    "mochawesome-merge": "^4.1.0",
    "mochawesome-report-generator": "^5.1.0"
```

**Comment by jennifer-shehane** [comment]
There's likely something misconfigured in the setup. Your mochawesome options should be `quiet` not `quite` by the way, but I don't think this is related. 

Issues in our GitHub repo are reserved for potential bugs or feature requests. This issue will be closed since it appears to be neither a bug nor a feature request.

We recommend questions relating to how to use Cypress be asked in our [community chat](https://gitter.im/cypress-io/cypress). Also try searching our [existing GitHub issues](https://gitter.im/cypress-io/cypress/issues), reading through our [documentation](https://docs.cypress.io), or searching [Stack Overflow](https://stackoverflow.com/questions/tagged/cypress) for relevant answers.

**Comment by shahavej** [comment]
hey, I am also facing this issue can anybody help

**Comment by jamilhf** [comment]
any updates on this issue ? were you able to solve it ?

**Comment by shahavej** [comment]
@jamilhf  still unable to merge all mocha reports. please do let me know if you have fix this issue :)

**Comment by jamilhf** [comment]
@shahavej  can you send me your configuration in cypress.json

**Comment by T-e-j-k-u-m-a-r** [comment]
Hi Team, 

I'm facing exactly the same issue. 

Here is the problem statement

ERROR: Failed to merge reports

Error: Pattern cypress/reports/mocha/*.json matched no report files

Is it still in open state ?

**Comment by T-e-j-k-u-m-a-r** [comment]
Hi Team, 

I'm getting merge error for mochawesome reports,   Please suggest the solution

ERROR: Failed to merge reports

Error: Pattern cypress/reports/mocha/*.json matched no report files

Here is my package.json file:

{
  "name": "Project1",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "clean:reports": "mkdir cypress\\reports\\ && mkdir cypress\\reports\\mocha\\ && mkdir cypress\\reports\\mochareports\\ ",
    "pretest": "npm run clean:reports",
    "scripts": "cypress run --spec 'cypress/integration/CafeTownSend/*' --headless",
    "merge-report": "mochawesome-merge  cypress/reports/mocha/*.json > cypress/reports/report.json",
    "generate-report": "marge cypress/reports/mochareports/report.json -f report -o cypress/reports/mochareports",
    "posttest": "npm run merge-report && npm run generate-report",
    "test": "npm run scripts || npm run posttest"
  },
  "keywords": [],
  "author": "Tejkumar Kempaiah",
  "license": "ISC",
  "devDependencies": {
    "cy-verify-downloads": "0.0.4",
    "cypress": "^8.1.0",
    "cypress-downloadfile": "^1.2.1",
    "cypress-file-upload": "^5.0.8",
    "cypress-iframe": "^1.0.1",
    "cypress-plugin-tab": "^1.0.5",
    "cypress-xpath": "^1.6.2",
    "mocha": "^9.0.3",
    "mochawesome": "^6.2.2",
    "mochawesome-merge": "^4.2.0",
    "mochawesome-report-generator": "^5.2.0"
  }
}


-------------------
Here is my cypress.json file:

{
  "projectId": "18xfp2",
  "$schema": "https://on.cypress.io/cypress.schema.json",
  "env": {
    "appurl": "https://trytestingthis.netlify.app/",
    "username": "test",
    "password": "test"
  },
  "watchForFileChanges": false,
  "defaultCommandTimeout": 10000,
  "pageLoadTimeout": 1000,
  "baseUrl": "https://skpatro.github.io/demo/signup/",
  "chromeWebSecurity": false,
  "reporter": "mochawesome",
  "reporterOptions": {
    "files": "mochawesome-report",
    "overwrite": false,
    "html": false,
    "json": true
  }
}
-------------------------

**Comment by denisbalyko** [comment]
You can change the legacy command:

from:
`mochawesome-merge  cypress/reports/mocha/*.json > cypress/reports/report.json`

to:
`mochawesome-merge  cypress/reports/mocha/*.json -o cypress/reports/report.json`

**Comment by SandipUmapEcotech** [comment]
"merge-report": "mochawesome-merge cypress/reports/mocha/.json -o cypress/reports/report.json",
"generate-report": "marge cypress/reports/mochareports/report.json -f report -o cypress/reports/mocha",

this worked for me just change the folder of final merged report

**Comment by aswine** [comment]
Pattern StagingJsonReports/.jsons/*.json matched no report files

I have my tests running as

npx cypress run --browser chrome --headless --spec cypress/integration/uk/smoke/*.spec.js --env grepTags=@trynow  --reporter-options reportDir=StagingJsonReports,reportFilename=StagingTests,html=false,json=true,overwrite=false 

After the execution I have my json reports for each individual tests in the folder "StagingJsonReports", but the execution fails saying StagingJsonReports/.jsons/*.json matched no report files. Is there a way to override this. Also my screenshots are getting copied from the default directory to the this custom report directory. Are there ways to override this.

**Comment by amoy180180** [comment]
> Pattern StagingJsonReports/.jsons/*.json matched no report files
> 
> I have my tests running as
> 
> npx cypress run --browser chrome --headless --spec cypress/integration/uk/smoke/*.spec.js --env grepTags=@TryNow --reporter-options reportDir=StagingJsonReports,reportFilename=StagingTests,html=false,json=true,overwrite=false
> 
> After the execution I have my json reports for each individual tests in the folder "StagingJsonReports", but the execution fails saying StagingJsonReports/.jsons/*.json matched no report files. Is there a way to override this. Also my screenshots are getting copied from the default directory to the this custom report directory. Are there ways to override this.

I have exactly the same issue, seems mochawesome is trying to merge the json files from .json folder, while the actual json files are created in the upper level folder.

**Comment by fragsman** [comment]
I would like to add some help. I struggled with this error too. 

From this:
`"mochawesome:merge": "npx mochawesome-merge 'cypress/results/mochawesome/*.json' -o mochawesome.json && npx marge mochawesome.json"`

To this:
`"mochawesome:merge": "npx mochawesome-merge cypress/results/mochawesome/*.json -o mochawesome.json && npx marge mochawesome.json"`

I just removed the quotation marks.. Funny thing. I had another project (with mochawesome-merge) 4.3.0 with the exact set up and it does work with the quotation mark. The one I had problems was Mochawesome-merge 5.0.0. No idea if that was the problem
